src/main/webapp/app/view/invitation/accept-invitation.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnDestroy } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { InvitationService } from 'app/entities/invitation/invitation.service';
+import { filter, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { AccountService } from 'app/core/auth/account.service';
+import { Subject } from 'rxjs';
+
+@Component({
+  selector: 'app-accept-invitation',
+  templateUrl: './accept-invitation.component.html',
+})
+export class AcceptInvitationComponent implements OnDestroy {
+  token?: string;
+  variant: 'login' | 'register';
+
+  private destroy$ = new Subject<void>();
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private invitationService: InvitationService,
+    private router: Router,
+    private accountService: AccountService
+  ) {
+    this.variant = activatedRoute.snapshot.data.variant;
+    this.activatedRoute.params
+      .pipe(
+        tap(params => (this.token = params['token'])),
+        switchMap(() => this.invitationService.checkTokenValidity(this.token!)),
+        tap(isTokenValid => {
+          if (!isTokenValid) {
+            this.router.navigate(['/invalid-token']);
+            return false;
+          }
+          return true;
+        }),
```

Comment:
theoretisch könntest du hier `isTokenValid` logisch mit `this.router.navigate(['/invalid-token'])` verbinden.

Reply:
ah der tap operator reicht den vorherigen wert weiter so wie ichs verstanden habe. dann muss man hier gar nichts return und es reicht einfach wenn man if (!isTokenValid) {this.router.navigate .. } schreibt.